sensor: vcnl36825t: fix: Address CID 347083

ubieda opened the pull request:
Coverity found this legitimate issue: the datasheet specifies PS_IT being 2-bits long (1, 2, 4, 8) and this driver assumes more steps are available. Remove extraneous fields so the Proximity integration setting fits in the expected 16-bit value.

Fixes #69125

ubieda committed:
sensor: vcnl36825t: fix: Address CID 347083

Coverity found this legitimate issue: the datasheet specifies PS_IT
being 2-bits long (1, 2, 4, 8) and this driver assumes more steps are
available. Remove extraneous fields so the Proximity integration
setting fits in the expected 16-bit value.

Signed-off-by: Luis Ubieda <luisf@croxel.com>

ubieda commented:
CC: @deveritec-rosc

deveritec-rosc commented:
Thank you!

Haven't seen this, might be I was in a "wrong" sensor. Might be a good idea to switch to an integer-based enum, then.

deveritec-rosc commented on drivers/sensor/vishay/vcnl36825t/vcnl36825t.c:
```diff
@@ -242,26 +242,10 @@ static int vcnl36825t_init_registers(const struct device *dev)
 		reg_value |= VCNL36825T_PS_IT_1T;
 		data->meas_timeout_us *= 1 * VCNL36825T_FORCED_FACTOR_SCALE;
```
You may remove `VCNL36825T_FORCED_FACTOR_SCALE` since it was necessary to get the 0.5 values into the integer range. If you do so, you can also simplify `VCNL36825T_FORCED_FACTOR_SCALE * VCNL36825T_FORCED_FACTOR_SCALE` in line 380 to a simple `VCNL36825T_FORCED_FACTOR_SCALE`

ubieda replied:
Thanks. Addressed.

ubieda committed:
sensor: vcnl36825t: Scrub redundant scaling on meas_timeout_us

Confirming the Proximity Integration is an integer, we don't need to
pre-apply the Scaling factor to then be accounted for when the actual
meas_timeout_us is calculated.

Signed-off-by: Luis Ubieda <luisf@croxel.com>
